### ISSUE TITLE
Custom fan control stopped working after upgrade to macOS Sonoma 14.4 beta [M1/M2/M3 laptops]

### ISSUE THREAD

**Issue by kleuter** [crystalidea/macs-fan-control#733]
Currently we have reports that it happens on MacBookPro18.1 and MacBookPro18.2

18.2:

![image](https://github.com/crystalidea/macs-fan-control/assets/2600624/744a815f-3d15-4085-8c70-f30570aa1e54)
![image](https://github.com/crystalidea/macs-fan-control/assets/2600624/8028089d-ab68-4221-9caa-ad5454510695)

Report from 18.1

> Good morning everyone, for a few days the program no longer controls the fans as before, I can identify this problem since I have installed Sonoma Beta Version of 14.4 (23E5180j)
> 
> Some idea how to solve?

**Comment by kleuter** [comment]
Can you try to do some CPU intensive task so the system will turn on the fans and then try to set some custom RPM value (e.g. Full Blast)?

**Comment by kleuter** [comment]
I'm get my hands on MacBookPro18,1 next week to check the issue

**Comment by Williamlml** [comment]
An issue for MacBookPro18,3 as well. 
<img width="933" alt="Screenshot 2024-02-02 at 15 17 19" src="https://github.com/crystalidea/macs-fan-control/assets/110677604/eaa3a1de-577c-4110-8cee-fdd92c4eb314">

**Comment by uknwnth** [comment]
same with 19,1 even after roll back to 14.3
<img width="841" alt="Screenshot 2024-02-02 at 11 34 50 PM" src="https://github.com/crystalidea/macs-fan-control/assets/14957336/afb38de7-f9ee-4696-83d4-0f2cd5826e43">

**Comment by uknwnth** [comment]
> same with 19,1 even after roll back to 14.3 <img alt="Screenshot 2024-02-02 at 11 34 50 PM" width="841" src="https://private-user-images.githubusercontent.com/14957336/301965413-afb38de7-f9ee-4696-83d4-0f2cd5826e43.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY5MTM1NjUsIm5iZiI6MTcwNjkxMzI2NSwicGF0aCI6Ii8xNDk1NzMzNi8zMDE5NjU0MTMtYWZiMzhkZTctZjllZS00Njk2LTgzZDQtMGYyY2Q1ODI2ZTQzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjAyVDIyMzQyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFkMzJkZmE1ZjU2ZDE5ZTA1MjgzYWY5ZjRlZmE2ZDc5Y2FmNjJkZjE1MzQzYmYwMWY5ODk4ZDM4YjY3ZjI3NjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.PJHGVyNej070BB9jOc0TJ9tnShMUl_GJq6eVZTjy8mI">

if fans are start spinning by system under heavy load tasks for some reason full blast and custom profile start to work again

**Comment by MysticalOS** [comment]
So I'm gonna guess they ported what was happening on specific M3 models to the rest of them in 14.4? IE it use curve but only when apple decides to let fans turn on?

**Comment by kleuter** [comment]
@uknwnth please post a screenshot where you set fans to Custom control but they are not spinning

**Comment by kleuter** [comment]
@MysticalOS looks like Sonoma 14.4 updates system firmware on Apple Silicon to version **10151.100.738.0.1** from **10151.81.1**

**Comment by MysticalOS** [comment]
At all? or just toggling fans on and off like firmware that was included on just 16 inch m3s before?

I've learned to live with not controlling when they turn on, but I don't accept apple turning off speed. apple keeps the M3 Max at 95C on default curve, mine keeps it at 70C.

there is no reason to keep it at 95C except to be utterly toxic and purposely keep your product from lasting. Apple should be called out for this.

**Comment by kleuter** [comment]
Not completely, but yes, seems like on new M3 models, will confirm this next week when I get my hands on 16" M1 Pro MBP

**Comment by MysticalOS** [comment]
Ok, so if M understanding right, they are basically backporting the change i've already dealt with, to everyone else. :\

Which is that fan kicks on whenever apple dang well pleases, but once it's on THEN we can control speed (up until apple turns it off again). I remember when M3 came out, this only occured on M3 16 inch models, 14 inch unaffected (well, was til now)

**Comment by suservu123** [comment]
i have same issue but if i run cinebench and the fan run. we can use macfancontrol to control speed like normal

**Comment by kleuter** [comment]
@suservu123 a demonstrating screenshot please? (screenshot where you set fans to Custom control but they are not spinning)

**Comment by uknwnth** [comment]
> @uknwnth please post a screenshot where you set fans to Custom control but they are not spinning

sure! I also post a video with the step-by-step process, but I found you already explain this behavior pretty well in this issue https://github.com/crystalidea/macs-fan-control/issues/731

https://youtu.be/rpvDDA4RD54

**Comment by kleuter** [comment]
@uknwnth thank you very much, Kiril!

**Comment by arkdev1** [comment]
Update:
I've installed update Sonoma Beta Version 14.4 (23E5191e), "problem" still present.

**Comment by DrakeJordan** [comment]
Facing the same issue here on Sonoma Beta 14.4 23E5191e, on a MBP 16" M2 Pro

**Comment by kleuter** [comment]
Looks like Apple decided to kill ~~our app ~~ custom fan control on M1/M2/M3 laptops

**Comment by arkdev1** [comment]
> Looks like Apple decided to kill ~our app~ custom fan control on M1/M2/M3 laptops

Seems like it. Tired of this "Apple moves"

**Comment by MysticalOS** [comment]
I wouldn't call it killed though. It still controls fan speed, just not on/off state no?. At least on my M3 Max, which has had the change much sooner than m1 and m2.

It still keeps it from overheating and throttling. yeah it won't keep temps from reaching 80C+ when they don't need to, cause apple waits to turn fans on til ages later, but yeah sucks can't control on/off state anymore :\

**Comment by KylieFever** [comment]
Hi,
As another data point, it’s the exact same situation on MacBookPro**18,3** models
RPM control does work but only once the fans have spun up

Also thanks for the work you’ve put into this app :)

**Comment by arkdev1** [comment]
> Hi, As another data point, it’s the exact same situation on MacBookPro**18,3** models RPM control does work but only once the fans have spun up
> 
> Also thanks for the work you’ve put into this app :)

As curiosity, the starts at 70°C? 80°C ?

**Comment by KylieFever** [comment]
> > Hi, As another data point, it’s the exact same situation on MacBookPro**18,3** models RPM control does work but only once the fans have spun up
> > Also thanks for the work you’ve put into this app :)
> 
> As curiosity, the starts at 70°C? 80°C ?

About to test this out. Did you have a certain sensor in mind? Or just the average CPU temp?

**Comment by KylieFever** [comment]
> > > > Hi, As another data point, it’s the exact same situation on MacBookPro**18,3** models RPM control does work but only once the fans have spun up
> > > > Also thanks for the work you’ve put into this app :)
> > > 
> > > 
> > > As curiosity, the starts at 70°C? 80°C ?
> > 
> > 
> > About to test this out. Did you have a certain sensor in mind? Or just the average CPU temp?
> 
> Average, thanks

No problem, fans turned on right as average hit 88.0ºC and stayed on until I reset control back to automatic

**Comment by MysticalOS** [comment]
I don't set back to automatic. i run a custom temp curve, and yeah i can't control when it turns on or off, apple does that, but it still keeps desired temp one it kicks in cause it seems apples turn on is between 80-90 and turn off is 65 or lower, i haven't measured exact point yet but it's def lower 60s upper 50s and that's good cause it keeps apple from turning my can curve off ONCE it kicks in.

**Comment by arkdev1** [comment]
Hi everyone, light at the end of the tunnel:

https://github.com/crystalidea/macs-fan-control/assets/138211005/f5baee40-26fe-4011-b643-639d599f15ee

Yesterday i've upgraded to Sonoma Beta 14.4 (23E5205c), and it works.

**Comment by MysticalOS** [comment]
And to confirm, this is while idle and not working a load? cause even on M3, and M1 and M2 in betas, you could still control fans if under a heavy load, just couldn't when not

**Comment by arkdev1** [comment]
> And to confirm, this is while idle and not working a load? cause even on M3, and M1 and M2 in betas, you could still control fans if under a heavy load, just couldn't when not

Exactly, under no load, i've powered on for work, and i've heard the fan full blast. I've tried 4/5 times before the video.

**Comment by kleuter** [comment]
@arkdev1 firmware version?

**Comment by arkdev1** [comment]
> @arkdev1 firmware version?

So, with `sw_vers`:
![Screenshot 2024-02-26 alle 13 14 20](https://github.com/crystalidea/macs-fan-control/assets/138211005/93011062-1a85-4f62-8e56-a3185b29aae1)

If you need other information please help me where to find.

**Comment by uknwnth** [comment]
> > @arkdev1 firmware version?
> 
> 
> 
> If you need other information please help me where to find.

you can find it here: System Information -> Hardware ->  System Firmware Version

<img width="1017" alt="Screenshot 2024-02-26 at 15 28 51" src="https://github.com/crystalidea/macs-fan-control/assets/14957336/8e4d6a50-590a-4f48-a5af-4e2a52d5c436">

**Comment by arkdev1** [comment]
@kleuter here the firmware version:

![Screenshot 2024-02-26 alle 13 33 13](https://github.com/crystalidea/macs-fan-control/assets/138211005/47677d6a-9388-493c-ab44-93bdf1cd05c4)


@uknwnth thanks!

**Comment by MysticalOS** [comment]
10151.101.3 for M3
seems to have reversed course as well. It's now letting sofware turn fans on too, not just adjust speed. so my can curve, set to kick on at exactly 50C, is actually kicking on at 50C and not at 80-90C like it was before

and of note, My M3 had fan control disabled from day 1, even prior to 14.4.
it always worked like this
software could not turn fans on or off, only set fan speed once they turned on
they turned on at apples descretion which was typically once it was super hot, but once on they'd stay on til it fell below like 55-60C

new to 14.4 latest betas, now letting software turn fan on and off too.

**Comment by kleuter** [comment]
OMG, great news! Will install it now

**Comment by uknwnth** [comment]
can confirm everything is back to normal with new firmware update 10151.101.3 (14.4 Beta 5) for m2 pro

**Comment by MysticalOS** [comment]
Update on this from further testing
https://github.com/crystalidea/macs-fan-control/issues/716#issuecomment-1974252178

**Comment by kleuter** [comment]
Duplicate of https://github.com/crystalidea/macs-fan-control/issues/785
